addons/mrp/models/mrp_production.py

```diff
@@ -1515,18 +1515,23 @@ def _get_consumption_issues(self):
                 expected_qty_by_product[move_product] += move_product_qty * order.qty_producing / order.product_qty
 
             done_qty_by_product = defaultdict(float)
+            issue_by_product = defaultdict(list)
             for move in order.move_raw_ids:
                 quantity = move.product_uom._compute_quantity(move.quantity, move.product_id.uom_id)
                 rounding = move.product_id.uom_id.rounding
                 if not (move.product_id in expected_qty_by_product or float_is_zero(quantity, precision_rounding=rounding)) or not move.picked:
-                    issues.append((order, move.product_id, quantity, 0.0))
+                    issue_by_product[move.product_id].append((order, move.product_id, 0.0, quantity))
```

Comment:
I don't think this can works well if we have more than one move_raw with the same product. 
e.g multiple workorder using the same component. instead why not setting `picked` value the same way we set `manual_consumption` here ?
https://github.com/odoo/odoo/blob/d64b4adf1e8fa1c414e163f117eaa87040875df8/addons/mrp/static/src/widgets/mrp_consumed.js#L26

Reply:
We already considered this situation, both `expected_qty_by_product` and `done_qty_by_product` summed up the qty for same product. We will only have 1 warning line for 1 product. I still updated the code to make it better to understand (hopefully :) )
Update `picked` value is already made for another task.